spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/filter/factory/RemoveJsonAttributesResponseBodyGatewayFilterFactory.java

```diff
@@ -93,22 +93,12 @@ public GatewayFilter apply(FieldListConfiguration config) {
 
 	private ObjectMapper mapper = new ObjectMapper();
 
-	private void removeJsonAttribute(JsonNode jsonBodyContent, List<String> fieldsToRemove, boolean deleteRecursively) {
-		if (deleteRecursively) {
-			for (JsonNode jsonNode : jsonBodyContent) {
-				if (jsonNode instanceof ObjectNode) {
-					((ObjectNode) jsonNode).remove(fieldsToRemove);
-					removeJsonAttribute(jsonNode, fieldsToRemove, true);
-				}
-				if (jsonNode instanceof ArrayNode) {
-					for (JsonNode node : jsonNode) {
-						removeJsonAttribute(node, fieldsToRemove, true);
-					}
-				}
-			}
+	private void removeJsonAttributes(JsonNode jsonNode, List<String> fieldNames, boolean deleteRecursively) {
+		if (jsonNode instanceof ObjectNode objectNode) {
+			objectNode.remove(fieldNames);
 		}
-		if (jsonBodyContent instanceof ObjectNode) {
-			((ObjectNode) jsonBodyContent).remove(fieldsToRemove);
+		if (deleteRecursively) {
+			jsonNode.forEach(childNode -> removeJsonAttributes(childNode, fieldNames, true));
```

Comment:
Modified the logic to recursively remove JSON nodes without type distinction, allowing recursive calls regardless of node type.